src/components/data-management/ProjectDetails.jsx

```diff
@@ -27,9 +29,50 @@ const ProjectDetails = ({ width, height }) => {
   );
   const [data, setData] = useState([]);
   const [sortedSpeciesData, setSortedSpeciesData] = useState([]);
+  const projects = useSelector((state) => state.projects);
+  const samples = useSelector((state) => state.samples);
   const activeProjectUuid = useSelector((state) => state.projects.meta.activeProject) || false;
   const activeProject = useSelector((state) => state.projects[activeProjectUuid]) || false;
 
+  const uploadFiles = (filesList, sampleType) => {
+    const samplesMap = filesList.reduce((acc, file) => {
+      const sampleName = file.name.trim().replace(/[\s]{2,}/ig, ' ').split('/')[0];
+      const sampleUuid = Object.values(samples).filter(
+        (s) => s.name === sampleName
+          && s.projectUuid === activeProjectUuid,
+      )[0]?.uuid;
+
+      return {
+        ...acc,
+        [sampleName]: {
+          ...acc[sampleName],
+          uuid: sampleUuid,
+          files: {
+            ...acc[sampleName]?.files,
+            [sampleName]: file,
+          },
+        },
+      };
+    }, {});
```

Comment:
Samples are first reduced to a grouping according to sample name, so that only 1 async dispatch to create a sample has to be run for each sample. Iterating over the files and checking if the sample has been created doesn't work because async dispatches hasn't yet completed when the next runs.